filemanager/s3manager_v2.go

```diff
@@ -333,3 +341,107 @@ func (l *s3ListSessionV2) Next() (fileObjects []*FileInfo, err error) {
 	}
 	return fileObjects, nil
 }
+
+func (m *s3ManagerV2) SelectObjects(ctx context.Context, selectConfig SelectConfig) (*SelectResult, error) {
+	client, err := m.getClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("s3 client: %w", err)
+	}
+
+	inputSerialization, outputSerialization, err := createS3SelectSerializationV2(selectConfig.InputFormat, selectConfig.OutputFormat)
+	if err != nil {
+		return nil, fmt.Errorf("error extracting input/output serialization: %w", err)
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, m.getTimeout())
+	selectObject, err := client.SelectObjectContent(ctx, &s3.SelectObjectContentInput{
+		Bucket:              aws.String(m.config.Bucket),
+		Key:                 aws.String(selectConfig.Key),
+		Expression:          aws.String(selectConfig.SQLExpression),
+		ExpressionType:      types.ExpressionTypeSql,
+		InputSerialization:  inputSerialization,
+		OutputSerialization: outputSerialization,
+	})
+	if err != nil {
+		cancel()
+		return nil, fmt.Errorf("error selecting object: %w", err)
+	}
+
+	stream := selectObject.GetStream()
+	if stream == nil {
+		cancel()
+		return nil, fmt.Errorf("error getting stream")
+	}
+	events := stream.Events()
+	if events == nil {
+		cancel()
+		return nil, fmt.Errorf("error getting events")
+	}
+	byteChan := make(chan []byte)
+	errorChan := make(chan error)
+	go func() {
+		defer func() {
+			if err := stream.Err(); err != nil {
+				errorChan <- err
+			}
+			close(byteChan)
+			close(errorChan)
+			stream.Close()
+			cancel()
+		}()
+		for {
+			select {
+			case <-ctx.Done():
+				if err := ctx.Err(); err != nil {
+					errorChan <- err
+				}
+				return
+			case event, ok := <-events:
+				if !ok {
+					return
+				}
+				switch e := event.(type) {
+				case *types.SelectObjectContentEventStreamMemberRecords:
+					byteChan <- e.Value.Payload
+				case *types.SelectObjectContentEventStreamMemberEnd:
+					return
+				}
+			}
+		}
+	}()
+	return &SelectResult{
```

Comment:
do we need two channel for this one?  There are two cases, either there could be a payload or error when reading. I am thinking we can do something like this
https://github.com/rudderlabs/sqlconnect-go/blob/a14c852eb9a8aba767f739e0df56ec4e653bf4c0/sqlconnect/async.go#L47